input/system/azure/system.go

```diff
@@ -0,0 +1,302 @@
+		if err != nil {
+			server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error making a Cosmos DB client: %v", err)
+			logger.PrintError("Azure/System: Failed to make a Cosmos DB client: %v\n", err)
+			return
+		}
+		// Search a server from the list
+		pager := clientFactory.NewClustersClient().NewListPager(nil)
+		for pager.More() {
+			page, err := pager.NextPage(ctx)
+			if err != nil {
+				logger.PrintError("Azure/System: Failed to advance page of Cosmos DB cluster list: %v\n", err)
+				break
+			}
+			for _, v := range page.Value {
+				if v.ID != nil {
+					resourceIDParts := strings.Split(*v.ID, "/")
+					resourceGroup := resourceIDParts[4]
+					resourceType := resourceIDParts[7]
+					resourceName := resourceIDParts[8]
+
+					if config.AzureDbServerName == resourceName {
+						system.Info.Azure = &state.SystemInfoAzure{
+							Location:                 util.StringPtrToString(v.Location),
+							CreatedAt:                util.TimePtrToTime(v.SystemData.CreatedAt),
+							State:                    util.StringCustomTypePtrToString(v.Properties.State),
+							SubscriptionID:           config.AzureSubscriptionID,
+							ResourceGroup:            resourceGroup,
+							ResourceType:             resourceType,
+							ResourceName:             resourceName,
+							CitusVersion:             util.StringPtrToString(v.Properties.CitusVersion),
+							HighAvailabilityEnabled:  util.BoolPtrToBool(v.Properties.EnableHa),
+							CoordinatorStorageMB:     util.Int32PtrToInt(v.Properties.CoordinatorStorageQuotaInMb),
+							NodeStorageMB:            util.Int32PtrToInt(v.Properties.NodeStorageQuotaInMb),
+							CoordinatorVCores:        util.Int32PtrToInt(v.Properties.CoordinatorVCores),
+							NodeVCores:               util.Int32PtrToInt(v.Properties.NodeVCores),
+							CoordinatorServerEdition: util.StringPtrToString(v.Properties.CoordinatorServerEdition),
+							NodeServerEdition:        util.StringPtrToString(v.Properties.NodeServerEdition),
+							NodeCount:                util.Int32PtrToInt(v.Properties.NodeCount),
+						}
+						resourceID = *v.ID
+						break
+					}
+				}
+			}
+		}
+	}
+
+	if resourceID == "" {
+		server.SelfTest.MarkCollectionAspectWarning(state.CollectionAspectSystemStats, "unable to find the database server info")
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the Reader permission of the database is granted to the managed identity.")
+		return
+	}
+
+	// Server metrics
+	client, err := azquery.NewMetricsClient(credential, nil)
+	if err != nil {
+		server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error making a metrics client: %v", err)
+		logger.PrintError("Azure/System: Failed to make a metrics client: %v\n", err)
+		return
+	}
+
+	// Query metrics data with 1 min interval, for the last 1 min (should return 1 value)
```

Comment:
I noticed that this API actually returns null values often for certain metrics. Read/Write throughputs with Flexible Server is a bad one I've seen, the following graph is from a test database but you can see that it's lacking values often (the dotted line is missing data):

![image](https://github.com/user-attachments/assets/fe243deb-cef3-4d28-84b2-2101966679f4)

I thought about changing the interval and timespan a bit to potentially make the data better quality (less null data), but I think expanding this is not going to help much. Also, the next available interval is 1m, 5m, 15m and no 10m, so I figured that I'll simply use 1m here.

Reply:
Hmm. Might be worth opening a ticket with Azure support, just to understand what's going on?

Reply:
I think we don't have a support line in Azure. Given that it's like this in Azure console too (above screenshot is from Azure console), I think they simply don't have the data, so I'd say it's okay to leave it as is.